public/app/features/query/state/DashboardQueryRunner/SnapshotWorker.ts

```diff
@@ -0,0 +1,42 @@
+import { Observable, of } from 'rxjs';
+import { cloneDeep } from 'lodash';
+import { AnnotationEvent } from '@grafana/data';
+
+import { DashboardQueryRunnerOptions, DashboardQueryRunnerWorker, DashboardQueryRunnerWorkerResult } from './types';
+import { emptyResult } from './utils';
+
+export class SnapshotWorker implements DashboardQueryRunnerWorker {
+  canWork({ dashboard }: DashboardQueryRunnerOptions): boolean {
+    const snapshots = dashboard.annotations.list.find((a) => a.enable && Boolean(a.snapshotData));
+    return Boolean(snapshots);
+  }
+
+  work(options: DashboardQueryRunnerOptions): Observable<DashboardQueryRunnerWorkerResult> {
+    if (!this.canWork(options)) {
+      return emptyResult();
+    }
+
+    const { dashboard } = options;
+    const dashAnnotations = dashboard.annotations.list.filter((a) => a.enable);
+    const snapshots = dashAnnotations.filter((a) => Boolean(a.snapshotData));
+    const annotations = snapshots.reduce((acc, curr) => {
+      return acc.concat(SnapshotWorker.translateQueryResult(curr, curr.snapshotData));
+    }, [] as AnnotationEvent[]);
+
+    return of({ annotations, alertStates: [] });
+  }
+
+  private static translateQueryResult(annotation: any, results: AnnotationEvent[]): AnnotationEvent[] {
```

Comment:
Can we type `annotation` with `AnnotationQuery<DataQuery>`?

Reply:
Will fix!

Reply:
Fixed